### PR TITLE
Refine setup and configuration docs

### PR DESCRIPTION
**Pull request by pellared**
## Why

Fixes https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/2071

Towards https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/2076

## What

- Add recommendation for configuration methods
- Update IIS instrumentation docs
- Update Windows Service instrumentation docs
- Add comments for Bash script
- Update PowerShell module docs

### CODE REVIEW

**Comment by pellared** [comment]
@ramanjaneyagupta PTAL and feel free to provide any feedback

File: docs/README.md [hunk 1]
```diff
@@ -149,27 +156,27 @@ and instrument your .NET application using the provided PowerShell module.
...
-Register-OpenTelemetryForWindowsService -WindowsServiceName "MyServiceName" -OTelServiceName "MyServiceDisplayName"
-
-# Setup environment to start instrumentation from the current PowerShell session
+# Setup the instrumentation for the current PowerShell session
 Register-OpenTelemetryForCurrentSession -OTelServiceName "MyServiceDisplayName"
 
-# Get current installation location
-Get-OpenTelemetryInstallDirectory
+# Run your application with instrumentation
+dotnet run
```
Comment (pellared) [inline docs/README.md:174]: I know that `dotnet run` is not currently working

@pjanotti Should we keep it or change the example to use some executable? E.g.

```suggestion
.\MyNetApp.exe
```

**Comment by pellared** [comment]
@theletterf PTAL. Lots of docs changes

File: docs/README.md (hunk 1 above)
Comment (pjanotti) [reply docs/README.md:174]: I will explicitly call out the current issues with `dotnet` CLI and suggest to create a shell to run the instrumented version. Basically: build and run `dotnet` CLI on shell __1__ then set up the environment for automatic instrumentation on shell __2__. A warning icon, plus, a link to the bug seems worthwhile at the moment.

**Review by pjanotti — COMMENTED** [review]
@pellared for the time being we need to ensure that the instrumentation environment variables are not set in the same context/shell in which `dotnet` CLI is being used. There are too many setting combinations and most of them lead to different problems. So explicitly call that out and recommend running the executable directly in a separate context/shell to get instrumentation going.

File: docs/iis-instrumentation.md [hunk 2]
```diff
@@ -1,27 +1,24 @@
 # Instrument an ASP.NET application deployed on IIS
 
-## Set environment variables
+## Setup
 
-You can add the [`<environmentVariables>`](https://docs.microsoft.com/en-us/iis/configuration/system.applicationhost/applicationpools/add/environmentvariables/)
-in `applicationHost.config`
-to set environment variables for given application pools.
+Use the `OpenTelemetry.DotNet.Auto.psm1"` PowerShell module
+to setup automatic instrumentation for IIS:
```
Comment (theletterf) [inline docs/iis-instrumentation.md:6]:
```suggestion
to set up automatic instrumentation for IIS:
```

File: docs/iis-instrumentation.md [hunk 3]
```diff
@@ -63,7 +60,30 @@ to set it for all ASP.NET application running in Integrated Pipeline Mode:
...
+to set environment variables for given application pools.
+
+> For IIS versions older than 10.0, you can consider creating a distinct user,
+  set its environment variables
+  and use it as the application pool user.
+
+You can consider setting common environment variables,
+for all application deployed to IIS
+by setting the environment variables for
+`W3SVC` and `WAS` Windows Services as described in [windows-service-instrumentation.md](windows-service-instrumentation.md).
```
Comment (theletterf) [inline docs/iis-instrumentation.md:89]:
```suggestion
Consider setting common environment variables,
for all applications deployed to IIS
by setting the environment variables for
`W3SVC` and `WAS` Windows Services as described in [windows-service-instrumentation.md](windows-service-instrumentation.md).
```

File: docs/windows-service-instrumentation.md [hunk 4]
```diff
@@ -1,49 +1,46 @@
 # Instrument a Windows Service running a .NET application
 
-We recommend setting environment variables for a Windows Service
-using the Windows Registry.
+## Setup
 
-> Remember to restart the Windows Service
-  after making changes to its configuration.
+Use the `OpenTelemetry.DotNet.Auto.psm1"` PowerShell module
+to setup automatic instrumentation for a Windows Service:
```
Comment (theletterf) [inline docs/windows-service-instrumentation.md:6]:
```suggestion
to set up automatic instrumentation for a Windows Service:
```

File: docs/windows-service-instrumentation.md [hunk 5]
```diff
@@ -1,49 +1,46 @@
...
 
-## Set environment variables for a Windows Service
+```powershell
+# Import the module
+Import-Module "OpenTelemetry.DotNet.Auto.psm1"
+
+# Install core files
+Install-OpenTelemetryCore
+
+# Setup your Windows Service instrumentation
```
Comment (theletterf) [inline docs/windows-service-instrumentation.md:15]:
```suggestion
# Set up your Windows Service instrumentation
```

File: docs/windows-service-instrumentation.md [hunk 6]
```diff
@@ -1,49 +1,46 @@
...
+in the Windows Registry.
 
 The registry key of a given Windows Service (named `$svcName`) is located under:
 
 ```powershell
 HKLM\SYSTEM\CurrentControlSet\Services\$svcName
 ```
 
-The environment variables can be defined
+The environment variables can are defined
```
Comment (theletterf) [inline docs/windows-service-instrumentation.md:39]:
```suggestion
The environment variables are defined
```

**Review by theletterf — CHANGES_REQUESTED** [review]

**Review by Kielek — APPROVED** [review]

File: docs/README.md (hunk 1 above)
Comment (pellared) [reply docs/README.md]: I will address it in a separate PR.

**Pull request merged**